syntax = "proto3";

option csharp_namespace = "IntelliCook.RecipeSearch.Client";

message ChatByRecipeRequest {
    int32 id = 1;
    string username = 2;
    string name = 3;
    repeated ChatByRecipeMessage messages = 4;
}

message ChatByRecipeResponse {
    ChatByRecipeMessage message = 1;
}

message ChatByRecipeStreamResponse {
    oneof response {
        ChatByRecipeStreamHeader header = 1;
        ChatByRecipeStreamContent content = 2;
    }
}

enum ChatByRecipeRole {
    USER = 0;
    ASSISTANT = 1;
}

message ChatByRecipeMessage {
    ChatByRecipeRole role = 1;
    string text = 2;
}

message ChatByRecipeStreamHeader {
    ChatByRecipeRole role = 1;
}

message ChatByRecipeStreamContent {
    string text = 1;
}
